backend/build_docs.py

```diff
@@ -16,134 +17,120 @@
-            )
-        elif 'yosys' in dst:
-            copied_files[dst_path] = src.split('/data/rtdocs/')[-1]
-        elif 'OpenSTA' in dst:
-            copied_files[dst_path] = opensta_docs_url
-        else:
-            copied_files[dst_path] = dst_path
-
-    return copied_files
-
+        update_src(src, dst_path)
 
-def copy_tree_track_src(src: str, dst: str) -> dict[str, str]:
-    copied_files = {}
 
+def copy_tree_track_src(src: str, dst: str) -> None:
     if not os.path.exists(src):
         logging.debug(f'Folder {src} does not exist. Exiting.')
         sys.exit(1)
 
     for root, _, files in os.walk(src):
         rel_path = os.path.relpath(root, src)
-        if rel_path == '.':
-            dst_dir = dst
-        else:
-            dst_dir = os.path.join(dst, rel_path)
+        dst_dir = dst if rel_path == '.' else os.path.join(dst, rel_path)
         os.makedirs(dst_dir, exist_ok=True)
 
         for file in files:
             src_file = os.path.join(root, file)
             dst_file = os.path.join(dst_dir, file)
-            shutil.copy2(src_file, dst_file)
 
-            dst_path = dst_file.split('backend/')[-1]
+            if os.path.exists(dst_file):
+                base, ext = os.path.splitext(file)
+                counter = 2
+                while os.path.exists(dst_file):
+                    new_file_name = f'{base}_{counter}{ext}'
+                    dst_file = os.path.join(dst_dir, new_file_name)
+                    logging.debug(
+                        f'File {dst_file} already exists. Renaming to {new_file_name}'
+                    )
+                    counter += 1
 
-            if 'OR_docs' in dst_file:
-                copied_files[dst_path] = (
-                    f"{or_docs_url}/{src_file.split('_sources/')[-1].replace('.md', '.html')}"
-                )
-            elif 'ORFS_docs' in dst_file:
-                copied_files[dst_path] = (
-                    f"{orfs_docs_url}/{src_file.split('_sources/')[-1].replace('.md', '.html')}"
-                )
-            elif 'manpages' in dst_file:
-                copied_files[dst_path] = (
-                    f"OpenROAD Manpages - {src.split('data/markdown/')[-1]}"
-                )
-            elif 'yosys' in dst_file:
-                copied_files[dst_path] = src.split('/data/rtdocs/')[-1]
-            elif 'OpenSTA' in dst_file:
-                copied_files[dst_path] = opensta_docs_url
-            else:
-                copied_files[dst_path] = dst_path
+            shutil.copy2(src_file, dst_file)
 
-    return copied_files
+            dst_path = dst_file.split('backend/')[-1]
+            update_src(src_file, dst_path)
 
 
 def clone_repo(url: str, folder_name: str, commit_hash: Optional[str] = None) -> None:
     target_dir = os.path.join(cur_dir, folder_name)
     logging.debug(f'Cloning repo from {url} to {target_dir}...')
-    command = f'git clone {url} --depth 1 {target_dir}'
+    command = f'git clone {url} {target_dir}'
```

Comment:
Remove `--depth 1` will mean a full clone for all branches. Is this what is needed?

Reply:
Not really, its removed now.

Reply:
What i mean is that you will be doing a full clone without depth 1. Is the runtime increased?

Reply:
I dont think its significant, but its not necessary. So I've removed it nevertheless